remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java

```diff
@@ -91,88 +75,104 @@ public ClientHandshaker(byte[] expectedTeeMeasurement) {
-    byte[] random = new byte[REPLAY_PROTECTION_ARRAY_LENGTH];
-    new Random().nextBytes(random);
-
-    // Create client hello message.
-    Message.ClientHello clientHello = new Message.ClientHello(random);
-
-    // Update current transcript.
-    byte[] serializedClientHello = clientHello.serialize();
-    appendTranscript(serializedClientHello);
-
-    state = state.transition();
-    return serializedClientHello;
   }
 
   /**
    * Responds to {@code ServerIdentity} message by creating a {@code ClientIdentity} message and
    * derives session keys for encrypting/decrypting messages from the server.
    * {@code ClientIdentity} message contains an ephemeral public key for negotiating session keys.
    *
-   * Transitions {@code ClientHandshaker} state from {@code Attesting} to {@code Attested} state.
+   * Transitions {@code ClientHandshaker} state from {@code State.ExpectingServerIdentity} to
+   * {@code State.Attested} state.
    */
   public byte[] processServerIdentity(byte[] serializedServerIdentity)
       throws IllegalStateException, IOException, GeneralSecurityException {
-    if (state != HandshakerState.ExpectingServerIdentity) {
-      throw new IllegalStateException(
-          "ClientHandshaker is not in the ExpectingServerIdentity state");
-    }
-
-    Message.ServerIdentity serverIdentity =
-        Message.ServerIdentity.deserialize(serializedServerIdentity);
-
-    // Update current transcript.
-    // Transcript doesn't include transcript signature from the server identity message.
-    ServerIdentity serverIdentityNoSignature = new Message.ServerIdentity(
-        serverIdentity.getEphemeralPublicKey(), serverIdentity.getRandom(),
-        serverIdentity.getSigningPublicKey(), serverIdentity.getAttestationInfo());
-    byte[] serializedServerIdentityNoSignature = serverIdentityNoSignature.serialize();
-    appendTranscript(serializedServerIdentityNoSignature);
-
-    // Verify server transcript signature.
-    byte[] transcriptHash = Hashing.sha256().hashBytes(transcript).asBytes();
-    SignatureVerifier verifier = new SignatureVerifier(serverIdentity.getSigningPublicKey());
-    if (!verifier.verify(transcriptHash, serverIdentity.getTranscriptSignature())) {
-      throw new VerifyException("Couldn't verify server transcript signature");
-    }
-
-    // Verify server attestation info.
-    if (!verifyAttestationInfo(serverIdentity.getAttestationInfo())) {
-      throw new VerifyException("Couldn't verify attestation info");
+    try {
+      if (state != State.ExpectingServerIdentity) {
+        throw new IllegalStateException(
+            "ClientHandshaker is not in the ExpectingServerIdentity state");
+      }
+
+      Message.ServerIdentity serverIdentity =
+          Message.ServerIdentity.deserialize(serializedServerIdentity);
+
+      // Update current transcript.
+      // Transcript doesn't include transcript signature from the server identity message.
```

Comment:
It's a part of the SC protocol.
It's not possible to include it, because it will be including `ServerIdentity` hash inside `ServerIdentity`

Reply:
Then I'd suggest updating the comment: 
```suggestion
      // Create a copy of serverIdentity but without the signature. Because the transcript must not include transcript signature from the server identity message.
```